src/proto/FunctionRpc.proto

```diff
@@ -314,7 +314,9 @@ message BindingInfo {
   // Direction of the given binding
   Direction direction = 3;
 
-  DataType dataType = 4;
+  bool hasDataType = 4;
+
+  DataType dataType = 5;
```

Comment:
what if DataType was:
```
enum DataType {
  undefined = 0;
  string = 1;
  binary = 2;
  stream = 3;
```

the [default enum](https://developers.google.com/protocol-buffers/docs/proto3#default) is "0", so it will go to undefined if we don't pass anything to it

Reply:
Nit: could the naming convention of the enum fields be consistent with others in the file?  `Direction` starts with lowercase while the others start with uppercase, so there's already a disparity (obviously changing existing ones would be a breaking change, so I'm not advocating for that).  Google's protobuf guidelines call for "uppercase with underscores", but 🤷‍♂️.